Makefile

```diff
@@ -1,49 +1,102 @@
+	swiftformat --strict .
 
 .PHONY: clean
 clean:
 	rm -rf ./build
 
+# It turns out SKIP_INSTALL=NO and BUILD_LIBRARY_FOR_DISTRIBUTION=YES are essential
+# so that the archive contains a .framework output.
+#
+# We no longer use -sdk, we use -destination, as recommended by
+# See https://developer.apple.com/documentation/xcode/creating-a-multi-platform-binary-framework-bundle
 .PHONY: framework
-framework:
-	xcodebuild archive -sdk $(DEVICE_SDK) -workspace Authgear.xcworkspace -scheme Authgear-iOS -configuration Release -archivePath ./build/Release/$(DEVICE_SDK)/Authgear
-	xcodebuild archive -sdk $(SIMULATOR_SDK) -workspace Authgear.xcworkspace -scheme Authgear-iOS -configuration Release -archivePath ./build/Release/$(SIMULATOR_SDK)/Authgear
+framework: clean
+	xcodebuild archive \
+		-destination "generic/platform=iOS" \
+		-workspace Authgear.xcworkspace \
+		-scheme Authgear-iOS \
+		-configuration Release \
+		-archivePath ./build/Release/iOS/Authgear \
+		SKIP_INSTALL=NO \
+		BUILD_LIBRARY_FOR_DISTRIBUTION=YES
+
+	xcodebuild archive \
+		-destination "generic/platform=iOS Simulator" \
+		-workspace Authgear.xcworkspace \
+		-scheme Authgear-iOS \
+		-configuration Release \
+		-archivePath ./build/Release/iOS_Simulator/Authgear \
+		SKIP_INSTALL=NO \
+		BUILD_LIBRARY_FOR_DISTRIBUTION=YES
 
+# See https://developer.apple.com/documentation/xcode/creating-a-multi-platform-binary-framework-bundle
 .PHONY: xcframework
-xcframework:
+xcframework: framework
 	xcodebuild -create-xcframework \
-		-framework ./build/Release/$(DEVICE_SDK)/Authgear.xcarchive/Products/Library/Frameworks/Authgear.framework \
-		-framework ./build/Release/$(SIMULATOR_SDK)/Authgear.xcarchive/Products/Library/Frameworks/Authgear.framework \
+		-archive ./build/Release/iOS/Authgear.xcarchive -framework Authgear.framework \
+		-archive ./build/Release/iOS_Simulator/Authgear.xcarchive -framework Authgear.framework \
 		-output ./build/Release/Authgear.xcframework
 
-.PHONY: build
-build: framework
-	xcodebuild build -sdk $(SIMULATOR_SDK) -workspace Authgear.xcworkspace -scheme 'iOS-Example'
-
 .PHONY: test
 test:
-	xcodebuild -workspace Authgear.xcworkspace -scheme Authgear-iOS -destination ${TEST_DESTINATION} test
+	xcodebuild test \
+		-destination $(TEST_DESTINATION) \
+		-workspace Authgear.xcworkspace \
+		-scheme Authgear-iOS
+
+.PHONY: archive
+archive:
+	xcodebuild archive \
+		-destination "generic/platform=iOS" \
+		-workspace ./example/ios_example.xcworkspace \
+		-scheme iOS-Example \
+		-configuration Release \
+		-archivePath $(ARCHIVE_PATH)
+
+.PHONY: exportArchive
+exportArchive:
+	xcodebuild -exportArchive \
+		-archivePath $(ARCHIVE_PATH) \
+		-exportOptionsPlist ./example/ExportOptions.plist \
+		-exportPath $(EXPORT_PATH)
+
+.PHONY:	validate-app
+validate-app:
+	xcrun altool --validate-app \
+		--file $(IPA_PATH) \
+		--type ios \
+		--apiKey $(API_KEY) \
+		--apiIssuer $(API_ISSUER)
+
+.PHONY: upload-app
+upload-app:
+	xcrun altool --upload-app \
```

Comment:
Oh I thought we are going to use fastlane?
https://www.notion.so/oursky/Android-4feac6f5d9ca4f8ea8f00c107547c7f5?pvs=4#366ca1c586454bce9ada65470b9add6d